frontendExternalSystem/src/main/java/com/virtuslab/gitmachete/frontend/externalsystem/project/MacheteProjectResolver.java

```diff
@@ -24,6 +31,20 @@
       @Nullable MacheteExecutionSettings settings,
       ExternalSystemTaskNotificationListener listener)
       throws ExternalSystemException, IllegalArgumentException, IllegalStateException {
+
+    var graphTableProvider = Option.of(settings)
+        .map(s -> s.getProject().getService(GraphTableProvider.class));
+    var hasSelectedGitMacheteTab = Option.of(settings)
+        .map(s -> s.getProject())
+        .map(this::hasSelectedGitMacheteTab)
+        .getOrElse(Boolean.FALSE);
```

Comment:
Wow how come I've never seen that 😯

Reply:
(Most likely I was just always relying on autoboxing for booleans... also `Boolean` is relatively rarely a type param in any generic 🤔 )